### PR TITLE
Solves #12 Added Forgot Password/ Reset Password functionality

### PR DESCRIPTION
**Pull request by AyonPal**
- Added forget API that send forget password email

- Added Reset API takes new password and id from mail to reset user password

- Added postman collections in API docs

### CODE REVIEW

File: APIs/controllers/authCtrl.js [hunk 1]
```diff
@@ -225,8 +225,175 @@ async function register(req, res) {
...
 }
+function reset(req, res){
+    if (!req.body.password) {
+        res.status(500).json({
+            success: false,
+            message: "password is required"
+        });
+        return;
+    }
+    var reset_token = req.body.reset_token;
```
Comment (AsishRaju) [inline APIs/controllers/authCtrl.js:246]: @AyonPal please add a check for `req.body.reset_token` as it's needed to make a DB query

File: APIs/controllers/authCtrl.js [hunk 2]
```diff
@@ -225,8 +225,175 @@ async function register(req, res) {
...
+                            success: false,
+                            message: err.message
+                        });
+                    }
+                    if (data){
+                        const html = `Hi there,
+        <br/>
+        Go to the Link below to Reset Your Password! Link valid for 30 minutes.
+        <br/><br/>
+        Please verify your email <a href="http://localhost:4000/api/v1/reset?id=${data.reset_token}">HERE</a>
```
Comment (AsishRaju) [inline APIs/controllers/authCtrl.js]: @AyonPal hey buddy!
The link your sending in the mail won't be valid as your reset controller is looking for both `reset_token & new_pass` in the body, as I can see you're sending a direct API request with `reset_token` in parameters, instead, send the reset_token as a text and link to a frontend page (not yet made) where user can paste his `reset_token & new_pass` and get his password reset.

**Review by AsishRaju — CHANGES_REQUESTED** [review]
@AyonPal Do check the review I have made and do the necessary changes ✌🏻

File: APIs/controllers/authCtrl.js (hunk 2 above)
Comment (AyonPal) [reply APIs/controllers/authCtrl.js]: Should i set the token like 6 digit numeric or something  in the email?

**Review by AsishRaju — APPROVED** [review]
Cool man, thanks for the PR 👍🏻

**Pull request merged**